### PR TITLE
Created Divide and Conquer Solution to "Kattio: Closest Pair"

### PR DESCRIPTION
**Pull request by dutinmeow**

### CODE REVIEW

**Comment by dutinmeow** [comment]
I'll add the code if needed and the other solution if I have time

**Comment by dutinmeow** [comment]
Also, shouldn't the other solution say "sweep line " instead of "set"?

File: content/5_Plat/Sweep_Line.mdx [hunk 1]
```diff
@@ -36,9 +36,24 @@ number line?) never explicitly mentioned in module
 
 <FocusProblem problem="closest" />
 
-### Solution 1
+<Spoiler title="Solution 1">
```
Comment (dolphingarlic) [inline content/5_Plat/Sweep_Line.mdx:39]: No need to mark as a spoiler

File: content/5_Plat/Sweep_Line.mdx [hunk 2]
```diff
@@ -36,9 +36,24 @@ number line?) never explicitly mentioned in module
...
+
+$\delta$ is the upperbound of the answer. If a more optimal answer exists, it must bridge the two halves of the array (ie one of its endpoints is in $L$ and the other is in $R$). Let $mx$ be the x-coordinate of any median of $P$. Define two sets $L'$ and $R'$ such that $L' = {p_i | x_i \leq mx, mx - x_i \leq \delta}$ and $R' = {p_i | x_i \geq mx, x_i - mx \leq \delta}$
+
+A brute force matching algorithm that computes $
+\texttt{dist}(p, q)$ for all $p \in L'$, $q \in R'$ would have a worst-case runtime of $(\frac n 2)^2 = \mathcal{O}(n^2)$ (recall that $L'$ and $R'$ may have up to $\frac N 2$ points). However, because we are searching for distances of at most $\delta$, it suffices for each $p \in L'$ to check all points ${q | q \in R', p.y - \delta \leq q.y \leq p.y + \delta}$. 
+
+It can be shown that for each point $p$, there is $\mathcal{O}(1)$ points that satisfy this property. Because each point in $R'$ is at least $\delta_r \geq \delta$ apart, arranging the points in the worst case would result in 6 points in the corners and sides of the bounding rectangle.
```
Comment (dolphingarlic) [inline content/5_Plat/Sweep_Line.mdx:50]: > It can be shown that

Can you show this?

File: content/5_Plat/Sweep_Line.mdx [hunk 3]
```diff
@@ -36,9 +36,24 @@ number line?) never explicitly mentioned in module
...
+
+To achieve the desired $\mathcal{O}(n)$  complexity per layer, we need to be able to efficiently get the points sorted by both x-coordinate (for dividing $P$) and y-coordinate (for matching between $L'$ and $R'$). This can be achieved by taking advantage of the merge-sort like algorithm. Sort by x-coordinate in the beginning, then for each step, merge the y-coordinates recursively. 
+
+Because each step now runs in linear time, by the mastern theorem $T(n) = 2T(n / 2) + \mathcal{O}(n)$, our solution now runs in $\mathcal{O}(n\log n)$ (note its similarities to merge sort).
```
Comment (dolphingarlic) [inline content/5_Plat/Sweep_Line.mdx:54]: > the mastern theorem

Do you mean the master theorem?

**Review by dolphingarlic — CHANGES_REQUESTED** [review]
There are also a bunch of grammar issues that you should fix

**Comment by bqi343** [comment]
Changes requested have not been made. Free free to create a new PR.

**Pull request closed without merging**